libs/utils/android/workloads/viewer.py

```diff
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from time import sleep
+
+from android import Screen, System
+from android.workload import Workload
+
+
+class Viewer(Workload):
+    """
+    Android generic Viewer workload
+    """
+    
+    # Let the system pick the best package
+    package = ''
+
+    def __init__(self, test_env):
+        super(Viewer, self).__init__(test_env)
+        self._log = logging.getLogger('Viewer')
+        self._log.debug('Workload created')
+
+        # Set of output data reported by the browser
+        self.db_file = None
+
+    def run(self, out_dir, uri, workload, collect=''):
+        """
+        Run viewer workload
+
+        :param out_dir: Path to experiment directory where to store results.
+        :type out_dir: str
+
+        :param uri: The URI of the item to display
+        :type location_search: str
+
+        :param workload: Function pointer to the workload to run once item
+        is opened and being viewed.
+        :type workload: func
```

Comment:
The thing is, viewing an image or playing a video requires some additional logic anyway: you might want to make sure the image decoding is over, same for video playback (and you may even have to start the video itself). So all this workload should be seen as: "I open apps, but you have to tell me what to do once it's loaded". That could be nothing, that could be swiping, clicking, pressing buttons, etc.

The current setup also makes it really easy to quickly sketch up a workload and tweak it on the go. Having a subclass for each and every workload is contrary to that. I agree that some complex/specific uses of the Viewer could be subclasses, but I don't think that should apply to all of them.